### PR TITLE
Xray Mutation can no longer be obtained via combining Thermal and Radioactive.

### PR DESCRIPTION
**Pull request by KathyRyals**
## About The Pull Request
Removes the Thermal + Radioactive combination, thus making the Xray mutation unobtainable in normal gameplay.

## Why It's Good For The Game

Xray mutation has always been an extremely good mutation, and an amazing powergaming tool. It's existence, combined with the fact that geneticists can find it in under five minutes and will often put it on public console for anyone to grab, is extremely detrimental to the balance of the game, because once everyone on the crew has grabbed the mutation, it becomes basically impossible to do anything stealthy as antagonist, which can lead to shitty situation where said antagonist elects to do nothing all shift, for fear of getting caught.

I believe the fog of war is one of the most important part of any top down game, and the fact that anyone can remove it by simply jabbing themselves with a mutator that spits one out every thirty seconds is detrimental to the overall balance of the game.

Xray eyes can still be implanted via surgery, but those needs ressources, researchs, and another player to actually do the surgery.

I'm hoping that a decrease in the amount of people with access to Xray will lead to antagonists feeling more free to do what they are supposed to do : antagonize the crew (especially true for Manuel).

## Changelog
:cl:
del: The Xray mutation can no longer be obtained via combining radioactive and thermal eyes.
/:cl:

### CODE REVIEW

**Comment by anconfuzedrock** [comment]
this does make me consider that perhaps the fact that xray exists instead encourages smart traitors to act before xray becomes available or to try to disable genetics. however, we don't have many smart traitors.

**Comment by KathyRyals** [comment]
Competent geneticists can find xray mutations in five minutes. Doesn't leave much room for the traitors to act, don't you think?

**Comment by anconfuzedrock** [comment]
Could say something similar about toxins bombs. At least for traitors they can deal with it as quickly as possible while the crew can't as easily go into toxins and disposals all the ttvs with no excuse.

**Comment by Iatots** [comment]
Thermals is more than enough to tryhard validhunt. Just chase random people in maint and when they stop, keep examining them until they hold an incriminating item.

**Comment by anconfuzedrock** [comment]
Thinking about it, this might be good for now, but now I'm imagining xray as a somewhat harder to get or distribute mutation could be useful as a motivator for antagonists. have been issues with antags doing literally nothing for a long time, but if xray (or probably mutations in general) took more work to get and kept their power then it might motivate antagonists to act and disable genetics if they want to avoid being tracked as easily. If the antag doesn't want to do that then they just have to figure out some other way to be subtle, maybe using stuff like the stealth implant and bluespace body bags.

Either way most of the reasons xray can be hard to deal with are also issues with thermal, so you might want to nag that too.

**Comment by Kaffe-work** [comment]
x-ray is especially problematic for AI traitors. however, I think this is more about an underlying issue in genetics rather than x-ray being a problem.

**Comment by MaximeVignals** [comment]
I also believe that genetics needs a rework. The aim of this PR is to nerf it a bit to wait for the full on rework.

**Comment by trexter555** [comment]
iMpRoVe dOnT rEmOvE

**Comment by KathyRyals** [comment]
I also believe that something is wrong with genetics. The goal of my PR is to curb genetics power while waiting for the rework.

**Comment by KathyRyals** [comment]
Woopsed

**Comment by ExcessiveUseOfCobblestone** [comment]
bruh moment

**Comment by Time-Green** [comment]
You should remove Thermals aswell, it’s almost just as bad as x-ray.

**Comment by ShibaInuLord** [comment]
bro....

**Comment by ExcessiveUseOfCobblestone** [comment]
should just make it a considerable effort to obtain.

someone should explain how people are rushing the genetics "tree", I have played it like twice since it got moved from medical so i don't know (or care in terms of a >playing) the latest strats to it.

**Comment by KathyRyals** [comment]
I don't think thermal is /that/ bad. I could remove it if enough people want it gone but I don't think removing both at the same time will ever get merged.

**Comment by Pugie** [comment]
Honestly x-ray is horrifically balanced, especially so combined with lategame xray laser guns and cakewalks most threats. Removing all non-antag forms of wallhacks is probably the best solution.

**Comment by KathyRyals** [comment]
Even nukies have to spend 10 TCs for Xray implant. So it's even worse that geneticists can just give it to everyone in 10 minutes.

**Comment by KathyRyals** [comment]
> should just make it a considerable effort to obtain, even better if it was not guarantee-able in a round.

I left the code for xray on purpose because I think you can find some xray mutators in some space ruins or something. Also it lets admins spawn them for Tc trades/ gimmicks.

As soon as you get your hands on someone with the mutation, you can then mass produce it for the shift.

@ExcessiveUseOfCobblestone

**Comment by Timberpoes** [comment]
> 
> 
> should just make it a considerable effort to obtain, even better if it was not guarantee-able in a round.
> 
> someone should explain how people are rushing the genetics "tree", I have played it like twice since it got moved from medical so i don't know (or care in terms of a >playing) the latest strats to it. I did learn that snowballs have damage though so that was interesting!

The current iteration of Time-Green's Genetics has been historically balanced around 1 thing - The HTML interface being so horrible and clunky to use that you can't do anything quickly. High ping genetics was a horrible, laggy mess while you waited for the HTML interface to give you an input box.

My tgui variant of the DNA Console interface is simply more responsive for solving. As a result, brute-forcing is faster.

Example of a double-XX bruteforce with the new interface with optimum solving method (No repeats, single-click on every gene)
jlBJfOCBdk

Example of a triple-XX bruteforce with a worst-case scenario of brute-forcing over 75% of all possible combinations with a slightly ineffecient method (1 repeat of AT-AT-?? per solving cycle, single click on every gene)
nsAKqBpqdf

If you're experienced in genetics, you start to figure out rough categories of mutations based on the number of Xs. The number of Xs is of course fuzzied, however a mutation that is a sea of Xs is likely to be something like Two Left Feet, Acidic Flesh or Telekinesis.

On the other side of the coin, Radiation is usually a very easy mutation to solve and getting radiation requires wading through the sea of easy-to-solve, trash mutations until you finally unlock it.

However, none of that metaknowledge is required to excel. More people are playing genetics in general thanks to how accessible it is as a starting job with the improved interface. And they're learning new tricks and techniques of their own, and learning from others.

Examples of speeding up solving are as follows: Making your starting monkeys from cubes instead of robbing the monkey pen, then using the monkeys in the pen to try and solve more difficult mutations with the hand sequencer. Making a note of "unsolvable" gene sequences (Those too difficult to justify brute forcing, particularly where you don't have a matching mutation on any of your test subjects to fill in any XXs.) and using your notes to solve them if they appear later and finally that plague upon the station - Two highly skilled geneticists working together who can rip through the entire list of mutations in about 60-70% of the time a solo geneticist can.

**Comment by ATH1909** [comment]
I'm legitimately concerned that this is going to leave genetics with **very** few good, non-griefy powers to distribute. Take a look at this chart of positive/>0 instability cost mutations:
image

X-ray is one of the few actually good powers that genetics has left. Yes, it sometimes outs antags who are making maint bases, but that is one of the few meaningful things that genetics can still contribute to the station (the other one being pumping out space adaptation mutators).

**Comment by ATH1909** [comment]
Oh, I forgot internal matyrdom: I'm giving that one a rating of "holy fuck this is bad", because you're actively worse off with it than without it.

**Comment by optimumtact** [comment]
lets just remove genetics and come up with something else

**Comment by trerri** [comment]
based orange man is based. why keep genetics if the only balance it ever had was being slow as shit and the only good things it ever had were so insanely busted that they were only kept un-nerfed by the sad excuse for balance that is poopy ui?

**Comment by ATH1909** [comment]
> based orange man is based. why keep genetics if the only balance it ever had was being slow as shit and the only good things it ever had were so insanely busted that they were only kept un-nerfed by the sad excuse for balance that is poopy ui?

we could just add other ways to slow genetics down and/or nerf brute forcing, then add some more actually good mutations to the pool

maybe make changing letters irradiate people/monkeys more?

**Comment by Trumpetplaya288** [comment]
> 
> 
> > based orange man is based. why keep genetics if the only balance it ever had was being slow as shit and the only good things it ever had were so insanely busted that they were only kept un-nerfed by the sad excuse for balance that is poopy ui?
> 
> we could just add other ways to slow genetics down and/or nerf brute forcing, then add some more actually good mutations to the pool
> 
> maybe make changing letters irradiate people/monkeys more?

please please please make genetics about radmaxxing
also lazy and terrible fix bad PR

**Comment by GuillaumePrata** [comment]
Maybe having a chance of changing letters to mutate Monkeys into bad shit that tries to kill you, I know it might not be good for the game, but I do wish more jobs had real hazards like this for you to worry about in the middle of the process.
The risk reward balance of slowly getting mutations while caring for the monkey or rushing mutations as fast as you can while possibly mutating the poor monkey in a flesh eating, armblade swinging lavaland tier fauna would make genetics a lot more fun IMO.

**Comment by Ratvar** [comment]
ATH1909, Xray is a very griefy power to distribute - griefy against antags and all that want to fight them

**Comment by ATH1909** [comment]
> Xray is a very griefy power to distribute - griefy against antags and all that want to fight them instead of that bald table-climbing creature. Dwarfism too is griefy - you greytide into kitchen, bar, science, cargo and whatnot and that is it. Whole thing radiates cancer on meta level

excuse me what

**Comment by Kelenius** [comment]
Maybe the antags that get fucked by x-ray should go also get x-ray. I've never seen genetics refuse to give x-ray to someone who asked.

**Comment by KathyRyals** [comment]
Getting Xray does nothing to prevent getting ruined by Xray. What good does it do you to know that everyone saw you ? In the end you still can't do anything stealthy.

**Comment by Pugie** [comment]
> Maybe the antags that get fucked by x-ray should go also get x-ray. I've never seen genetics refuse to give x-ray to someone who asked.

This is the issue... once easy access wallhacks are obtained everyone else needs them or they risk being totally obliterated by a person they can't have possibly known could even see them. Even when both sides have x-ray any possibility of evasion via stealth goes totally out the window.

Also just everyone having x-ray isn't much of a valid argument as there are a lot of simplemob antagonists & similar. I don't think xenomorph queen is a bad example of this because despite its own balancing issues, having some guy with x-ray and an x-ray laser gun killing you through a wall with no danger to themselves and literally zero counterplay that the queen can do is not fun.

**Comment by Iatots** [comment]
I remember way back when det's thermal goggles and benos were on the chopping block, the idea of having a new layer for infrared (and maybe xray) was thrown around, were people would have color blots around them based on temperature. It was scrapped I think because of game limitations with the number of layers. Have conditions changed to make this possible? I think I remember jelly blood layer being a thing, if that was removed there could be an open spot.

**Comment by ATH1909** [comment]
> > Maybe the antags that get fucked by x-ray should go also get x-ray. I've never seen genetics refuse to give x-ray to someone who asked.
> 
> This is the issue... once easy access wallhacks are obtained everyone else needs them or they risk being totally obliterated by a person they can't have possibly known could even see them. Even when both sides have x-ray any possibility of evasion via stealth goes totally out the window.
> 
> Also just everyone having x-ray isn't much of a valid argument as there are a lot of simplemob antagonists & similar. I don't think xenomorph queen is a bad example of this because despite its own balancing issues, having some guy with x-ray and an x-ray laser gun killing you through a wall with no danger to themselves and literally zero counterplay that the queen can do is not fun.

xenos get thermal vision, just sayin'

**Comment by Pugie** [comment]
> > > Maybe the antags that get fucked by x-ray should go also get x-ray. I've never seen genetics refuse to give x-ray to someone who asked.
> > 
> > 
> > This is the issue... once easy access wallhacks are obtained everyone else needs them or they risk being totally obliterated by a person they can't have possibly known could even see them. Even when both sides have x-ray any possibility of evasion via stealth goes totally out the window.
> > Also just everyone having x-ray isn't much of a valid argument as there are a lot of simplemob antagonists & similar. I don't think xenomorph queen is a bad example of this because despite its own balancing issues, having some guy with x-ray and an x-ray laser gun killing you through a wall with no danger to themselves and literally zero counterplay that the queen can do is not fun.
> 
> xenos get thermal vision (which lets them see mobs through walls), just sayin'

Thermal vision is just as poor for balance but at least makes some sense for xenos who are fairly limited in their choices of weaponry & their purpose being an ultimate form of predator that can pick people off, similar to a changeling.

**Comment by Learner10** [comment]
Make X-Ray an Ethereal gene trait, then no one will have it.

**Comment by ATH1909** [comment]
Like, I guess my point here is that if none of the genetics powers can be strong because gamers can (apparently) grab them all in 20 minutes, maybe we should nerf what's being used to grab most of the powers in 20 minutes instead of nerfing/removing the actually good powers themselves.

File: code/datums/mutations/_combined.dm [hunk 1]
```diff
@@ -17,10 +17,12 @@
 	required = "/datum/mutation/human/strong; /datum/mutation/human/radioactive"
 	result = HULK
 
+//Commented in an effort to keep the code for a potential later genetics rework maybe?
```
Comment (optimumtact) [inline code/datums/mutations/_combined.dm:20]: Don't do this, remove the code in question, we use git for a reason

Comment (KathyRyals) [reply code/datums/mutations/_combined.dm]: Done!

**Comment by ExcessiveUseOfCobblestone** [comment]
this can prob comeback when theres a fair balance mechanic behind it.

**Comment by Learner10** [comment]
Easiest way is probably to make a lot more mutations.

**Pull request merged**